src/app/client/src/app/modules/dashboard/components/filter/filter.component.spec.ts

```diff
@@ -187,10 +217,30 @@ describe('FilterComponent', () => {
     expect(component.selectedFilters).toEqual({
       'state': ['01285019302823526477']
     });
+    expect(component.filters).toEqual([{ controlType: 'multi-select',
+     displayName: 'Select state', 
+     options: [ '01285019302823526477', '013016492159606784174', '0130385861180866561', 'b00bc992ef25f1a9a8d', 'b00bc992ef25f1a9a8d63291e20efc8d' ], 
+     reference: 'state' }]);
+    
+  }));
+
+  it('should get filter data without selected filter', fakeAsync(() => {
+    component.ngOnInit();
+    tick(1000);
+    component.selectedFilters ={};
+    tick(1000);
```

Comment:
why do you need a 2000 ms delay here, since you're not doing any async tasks here?

Also, this test case doesn't have proper assertion. you're setting the value and asserting on the same hard-coded value. Please add a relevant test with respect to the use case of the function.

Reply:
resolved